packages/flutter_tools/lib/src/ios/code_signing.dart

```diff
@@ -116,229 +159,787 @@ Future<Map<String, String>?> getCodeSigningIdentityDevelopmentTeamBuildSetting({
+      final List<_ProvisioningProfile> validProvisioningProfiles = await _getProvisioningProfiles();
+      if (validProvisioningProfiles.isEmpty) {
+        _logger.printError(
+          'No provisioning profiles were found. To learn how to create or download '
+          'a provisioning profile, please see '
+          'https://developer.apple.com/help/account/manage-provisioning-profiles/create-a-development-provisioning-profile',
+          emphasis: true,
         );
+        _logger.printWarning(_codeSignSelectionCanceled);
+        return;
+      }
+      final _ProvisioningProfile? profile = await _selectProvisioningProfile(
+        validProvisioningProfiles,
+      );
+      if (profile == null) {
+        _logger.printWarning(_codeSignSelectionCanceled);
+        return;
       }
+      _saveProvisioningProfile(profile);
+    }
+  }
+
+  /// Prompt user to select a code-signing style (Automatic or Manual).
+  Future<_CodeSigningStyle?> _selectSigningStyle() async {
+    _logger.printStatus('Code Signing Styles:', emphasis: true);
+    _logger.printStatus(
+      '  This setting specifies the method used to acquire and locate signing '
+      'assets. Choose Automatic to let Xcode automatically create and update '
+      'profiles, app IDs, and certificates. Choose Manual to create and update '
+      'these yourself on the developer website.',
+    );
+    _logger.printStatus('[1]: ${_CodeSigningStyle.automatic.label} (recommended)');
+    _logger.printStatus('[2]: ${_CodeSigningStyle.manual.label}');
+    final String choice = await _terminal.promptForCharInput(
+      List<String>.generate(2, (int number) => '${number + 1}')..add('q'),
+      prompt: 'Select a signing style (or "q" to quit)',
+      defaultChoiceIndex: 0, // Just pressing enter chooses the first one.
+      logger: _logger,
+      displayAcceptedCharacters: false,
+    );
+    if (choice == '1') {
+      return _CodeSigningStyle.automatic;
+    } else if (choice == '2') {
+      return _CodeSigningStyle.manual;
+    }
+    return null;
+  }
+
+  /// Prompt user to select a code-signing identity from a list of [validCodeSigningIdentities].
+  /// Selects the first one found without prompting if there is no stdin or if
+  /// [autoSelectSingle] is true and only one identity was found.
+  ///
+  /// Throw an error if [throwOnCancel] is true and the user quits while
+  /// selecting an identity.
+  Future<String?> _selectSigningIdentity(
+    List<String> validCodeSigningIdentities, {
+    bool autoSelectSingle = false,
+    bool throwOnCancel = false,
+  }) async {
+    if (validCodeSigningIdentities.isEmpty) {
+      return null;
+    }
+
+    if (autoSelectSingle && validCodeSigningIdentities.length == 1) {
+      return validCodeSigningIdentities.first;
     }
 
     // If terminal UI can't be used, just attempt with the first valid certificate
     // since we can't ask the user.
-    if (!terminal.usesTerminalUi) {
+    if (!_terminal.stdinHasTerminal) {
```

Comment:
Could you expand when this can happen? I'm a little concerned that `_getCodeSigningBuildSettings` might auto-save this code signing identity without asking the user, but I don't fully understand this scenario.